components/bsx/specific/MultiPaymentFeeButton.vue

```diff
@@ -4,49 +4,43 @@
   </nuxt-link>
 </template>
 
-<script lang="ts">
+<script setup lang="ts">
 import { getAssetIdByAccount } from '@/utils/api/bsx/query'
-import { Component, Prop, Watch, mixins } from 'nuxt-property-decorator'
-import UseApiMixin from '@/utils/mixins/useApiMixin'
-import AssetMixin from '@/utils/mixins/assetMixin'
+
 import shouldUpdate from '@/utils/shouldUpdate'
 
-@Component({})
-export default class MultiPaymentFeeButton extends mixins(
-  AssetMixin,
-  UseApiMixin
-) {
-  @Prop({ type: String, required: false }) public accountId!: string
-  @Prop({ type: String, default: 'bsx', required: false })
-  public prefix!: string
-  protected tokenId = '0'
+const props = withDefaults(
+  defineProps<{
+    accountId: string
+    prefix?: string
+  }>(),
+  { prefix: 'bsx' }
+)
+const tokenId = ref('0')
+const asset = computed(() => useAsset().assetIdOf(tokenId.value))
 
-  get asset() {
-    return this.assetIdOf(this.tokenId)
-  }
+const unit = computed(() => asset.value.symbol)
+const url = computed(() => `/${props.prefix}/assets`)
 
-  get unit() {
-    return this.asset.symbol
-  }
-
-  get url(): string {
-    return `/${this.prefix}/assets`
-  }
-
-  async fetchCurrency() {
-    try {
-      const api = await this.useApi()
-      this.tokenId = await getAssetIdByAccount(api, this.accountId)
-    } catch (e) {
-      this.$consola.log(e)
-    }
+const fetchCurrency = async () => {
+  try {
+    const { apiInstance } = useApi()
+    const api = await apiInstance.value
+    tokenId.value = await getAssetIdByAccount(api, props.accountId)
+  } catch (e) {
+    console.log(e) // Use `console.log` instead of `this.$consola.log`
```

Comment:
Whats wrong with consola? just add `const { $consola } = useNuxtApp()`

Reply:
oh sorry @roiLeo  the comment was there originally. I did not add that and just migrated the code.

Do you want me to change it to this?

Reply:
Yes please